libenv/sysinfo.c

```diff
@@ -1645,6 +1659,95 @@ static void OSClasses(EvalContext *ctx)
 
 /*********************************************************************************/
 
+#ifdef __APPLE__
+static void Apple_Version(EvalContext *ctx)
+{
+    FILE *pp = NULL;
+
+    Log(LOG_LEVEL_VERBOSE, "This appears to be an apple system.");
+    Log(LOG_LEVEL_VERBOSE, "Looking for product name and version...");
+    if ((!FileCanOpen("/usr/bin/sw_vers", "r") || ((pp = cf_popen("/usr/bin/sw_vers", "r", true)) == NULL)))
+    {
+        Log(LOG_LEVEL_ERR, "Could not open and run /usr/bin/sw_vers to find macOS system version information.");
+        return;
+    }
+
+    size_t line_size = CF_BUFSIZE;
+    char *line = xmalloc(line_size);
+    int revcomps = 0, major, minor, patch;
+    char *flavor = NULL, *product_name = NULL, *r;
+
+    while (CfReadLine(&line, &line_size, pp) != -1)
+    {
+        if (STARTSWITH(line, "ProductName:"))
+        {
+            r = strrchr(line, '\t');
+            if (r == NULL || ++r == NULL)
+            {
+                continue;
+            }
+            product_name = SafeStringDuplicate(r);
+            ToLowerStrInplace(r);
+            flavor = SafeStringDuplicate(r);
+            EvalContextClassPutHard(
+                    ctx,
+                    r,
+                    "inventory,attribute_name=none,source=agent,derived-from=sw_vers");
+        }
+        else if (STARTSWITH(line, "ProductVersion:"))
+        {
+            r = strrchr(line, '\t');
+            if (r == NULL || ++r == NULL)
+            {
+                continue;
+            }
+            revcomps = sscanf(r, "%u.%u.%u", &major, &minor, &patch);
+        }
+    }
+
+    if (!flavor)
+    {
+        cf_pclose(pp);
```

Comment:
Will need to free `line` here as well
```suggestion
        cf_pclose(pp);
        free(line);
```